modules/integration/tests-integration/tests-server/src/test/java/org/wso2/mb/integration/tests/server/mgt/MetricsTestCase.java

```diff
@@ -0,0 +1,458 @@
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.MalformedURLException;
+import java.rmi.RemoteException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * These test will validate metrics functionality in message broker.
+ * This class has 4 test cases.
+ * 1. Metrics reporting functionality test case
+ * 2. Validate queue subscribers count metric
+ * 3. Validate topic subscribers count metric
+ */
+public class MetricsTestCase extends MBIntegrationBaseTest {
+
+	private String sessionCookie;
+
+	/**
+	 * Initializing test case
+	 *
+	 * @throws XPathExpressionException
+	 */
+	@BeforeClass(alwaysRun = true) public void init()
+			throws XPathExpressionException, MalformedURLException, AutomationUtilException,
+			       MetricsConfigException, MetricsLevelConfigException, RemoteException,
+			       LoginAuthenticationExceptionException {
+		super.init(TestUserMode.SUPER_TENANT_USER);
+
+		LoginLogoutClient loginLogoutClientForAdmin =
+				new LoginLogoutClient(super.automationContext);
+		sessionCookie = loginLogoutClientForAdmin.login("admin", "admin", "localhost");
+	}
+
+
+	/**
+	 * This test case will validate queue subscriber count metric.
+	 * Steps -
+	 * 1. Create 5 subscribers and start sending messages.
+	 * 2. Report metric.
+	 * 3. Validate metric.
+	 * @throws AndesClientConfigurationException
+	 * @throws JMSException
+	 * @throws NamingException
+	 * @throws IOException
+	 * @throws AndesClientException
+	 */
+	@Test(groups = "wso2.mb", description = "Queue subscribers count metric test case")
+	public void performQueueSubscriberCountTestCase()
+			throws AndesClientConfigurationException, JMSException, NamingException, IOException,
+			       AndesClientException, InterruptedException, AutomationUtilException,
+			       XPathExpressionException, MalformedObjectNameException {
+
+		long msgCount = 1000L;
+		int subscriberCount = 5;
+
+		// Creating a consumer client configuration
+		AndesJMSConsumerClientConfiguration consumerConfig =
+				new AndesJMSConsumerClientConfiguration(ExchangeType.QUEUE, "singleQueue");
+		consumerConfig.setMaximumMessagesToReceived(msgCount * 2);
+		consumerConfig.setPrintsPerMessageCount(msgCount / 10L);
+
+		// Creating a publisher client configuration
+		AndesJMSPublisherClientConfiguration publisherConfig =
+				new AndesJMSPublisherClientConfiguration(ExchangeType.QUEUE, "singleQueue");
+		publisherConfig.setNumberOfMessagesToSend(msgCount);
+		publisherConfig.setPrintsPerMessageCount(msgCount / 10L);
+
+		// Creating subscribers
+		AndesClient consumerClient = null;
+		for (int i = 0; i < subscriberCount; i++) {
+			consumerClient = new AndesClient(consumerConfig, true);
+			consumerClient.startClient();
+		}
+
+		// Creating publishers
+		AndesClient publisherClient = new AndesClient(publisherConfig, true);
+		publisherClient.startClient();
+
+		//Waiting for clients to start
+		Thread.sleep(5000);
```

Comment:
Can't a blocking "startClient" operation be implemented? .. rather than doing a thread sleep, like, how can we be sure, even with 5s, the client will be started? ..